File: src/backend/MasterMind.API/Models/Entities/Lead.cs
namespace MasterMind.API.Models.Entities;

/// <summary>
/// Lead entity for tracking inquiries and potential students
/// </summary>
public class Lead : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Mobile { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? ParentName { get; set; }
    public string? ParentMobile { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public LeadSource Source { get; set; }
    public string? SourceDetails { get; set; } // e.g., "Google Ads Campaign 1"
    public string? InterestedClass { get; set; }
    public string? InterestedSubject { get; set; }
    public LeadStatus Status { get; set; } = LeadStatus.New;
    public LeadPriority Priority { get; set; } = LeadPriority.Medium;
    public DateTime? NextFollowupDate { get; set; }
    public string? Remarks { get; set; }
    public int? AssignedToUserId { get; set; }
    public int? ConvertedStudentId { get; set; }
    public DateTime? ConvertedAt { get; set; }

    // Navigation properties
    public User? AssignedToUser { get; set; }
    public Student? ConvertedStudent { get; set; }
    public ICollection<LeadFollowup> LeadFollowups { get; set; } = new List<LeadFollowup>();
}

public enum LeadSource
{
    WalkIn,
    PhoneCall,
    Website,
    SocialMedia,
    Referral,
    Advertisement,
    Event,
    Other
}

public enum LeadStatus
{
    New,
    Contacted,
    Interested,
    FollowUp,
    Negotiation,
    Converted,
    Lost,
    NotInterested
}

public enum LeadPriority
{
    Low,
    Medium,
    High,
    Urgent
}

File: src/frontend/mastermind-web/src/types/lead.ts
export type LeadStatus = 'New' | 'Contacted' | 'Interested' | 'FollowUp' | 'Negotiation' | 'Converted' | 'Lost' | 'NotInterested'
export type LeadSource = 'Website' | 'PhoneCall' | 'Referral' | 'SocialMedia' | 'Advertisement' | 'WalkIn' | 'Event' | 'Other'
export type LeadPriority = 'Low' | 'Medium' | 'High' | 'Urgent'

export interface Lead {
  id: number
  name: string
  phone: string
  email: string
  parentName?: string
  parentMobile?: string
  address?: string
  city?: string
  source: string
  sourceDetails?: string
  status: string
  priority: string
  interestedClass?: string
  interestedSubject?: string
  nextFollowup?: string
  lastFollowup: string
  notes?: string
  assignedTo?: string
  convertedStudentId?: number
  convertedAt?: string
  createdAt?: string
  updatedAt?: string
  followupCount: number
}

export interface LeadFormData {
  name: string
  phone: string
  email?: string
  parentName?: string
  parentMobile?: string
  address?: string
  city?: string
  source?: string
  sourceDetails?: string
  interestedClass?: string
  interestedSubject?: string
  priority?: string
  nextFollowup?: string
  notes?: string
}

export interface FollowUpFormData {
  type?: string
  notes?: string
  status?: string
  nextFollowup?: string
  nextFollowupNotes?: string
}

export interface LeadFilters {
  search: string
  status: string
  source?: string
  page?: number
  pageSize?: number
}

export interface LeadStats {
  total: number
  new: number
  contacted: number
  interested: number
  followUp: number
  negotiation: number
  converted: number
  lost: number
}
